File: xposed/HookedApp/app/src/main/java/com/example/diman/hookedapp/MainActivity.java
package com.example.diman.hookedapp;

import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

  private RootChecker mRootDetector = new RootChecker();

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    if (mRootDetector.isDeviceRooted()) {
      showRootedDeviceDialog();
    }
  }

  private void showRootedDeviceDialog() {
    AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
    alertDialogBuilder.setTitle("Rooted Device!");
    alertDialogBuilder.setMessage("You are not allowed to use this app on a rooted device");
    alertDialogBuilder.setCancelable(false);
    AlertDialog alertDialog = alertDialogBuilder.create();
    alertDialog.show();
  }
}

File: repackaged/MyApp/app/src/main/java/com/dimanem/security/repackaged/example/myapp/MainActivity.kt
package com.dimanem.security.repackaged.example.myapp

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.app.AlertDialog

class MainActivity : AppCompatActivity() {

  private val BASE_URL = "https://www.example.com"

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    showBaseUrlDialog()
  }

  private fun showBaseUrlDialog() {
    val alertDialogBuilder = AlertDialog.Builder(this)
    alertDialogBuilder.setTitle("Base url")
    alertDialogBuilder.setMessage("$BASE_URL")
    val alertDialog = alertDialogBuilder.create()
    alertDialog.show()
  }
}
